File: cpp/epi/tree_preorder.cc
#include <memory>
#include <vector>
#include "../generic_types/binary_tree_node.h"
#include "generic_test.h"
using std::unique_ptr;
using std::vector;

vector<int> PreorderTraversalNaive(const unique_ptr<BinaryTreeNode<int>>& tree) {
  return {};
}

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::string func_name = "PreorderTraversalNaive";
  std::vector<std::string> param_names{"tree"};

  return GenericTestMain(args, "tree_preorder.cc", "../test_data/epi/tree_preorder.tsv",
                         &PreorderTraversalNaive, DefaultComparator{}, func_name, param_names);
}

File: java/epi/TreePreorder.java
package epi;

import generic_types.BinaryTreeNode;
import test_framework.EpiTest;
import test_framework.GenericTest;

import java.util.List;

public class TreePreorder {
  @EpiTest(testDataFile = "../test_data/epi/tree_preorder.tsv")
  public static List<Integer> preorderTraversalNaive(BinaryTreeNode<Integer> tree) {
    return null;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "TreePreorder.java",
                new Object() {
                }.getClass().getEnclosingClass())
            .ordinal());
  }
}
